api/cloud/oc_cloud_manager.c

```diff
@@ -223,33 +222,27 @@ _register_handler(oc_cloud_context_t *ctx, oc_client_response_t *data)
 
   cloud_store_dump_async(&ctx->store);
   ctx->retry_count = 0;
-  cloud_set_last_error(ctx, CLOUD_OK);
-
   ctx->store.status |= OC_CLOUD_REGISTERED;
-
-  cloud_set_cps(ctx, OC_CPS_REGISTERED);
-
-  return 0;
+  return err;
 
 error:
   ctx->store.status |= OC_CLOUD_FAILURE;
-  if (err == CLOUD_ERROR_CONNECT) {
-    if (!is_retry_over(ctx)) {
-      // While retrying, keep last error (clec) to CLOUD_OK
-      cps = OC_CPS_REGISTERING;
-      err = CLOUD_OK;
-    }
-  }
-  cloud_set_cps_and_last_error(ctx, cps, err);
-  return -1;
+  return err;
 }
 
 void
 oc_cloud_register_handler(oc_client_response_t *data)
 {
   cloud_api_param_t *p = (cloud_api_param_t *)data->user_data;
   oc_cloud_context_t *ctx = p->ctx;
-  _register_handler(ctx, data);
+  oc_cloud_error_t err = _register_handler(ctx, data);
+  switch (err) {
```

Comment:
Nitpick: The `switch` works, though `CLOUD_OK ` and `default`, alternative would be to add `bool` parameter to `_register_handler` to control whether `retry` should be run. However, 7 duplicated lines is not something we should lose sleep over, so I don't mind if it stays like this.

Reply:
I used the `retry` flag